### ISSUE TITLE
Container fails to start after v1.9.3 update - panic: constraint failed error

### ISSUE THREAD

**Issue by beastarman74** [Forceu/Gokapi#215]
On update to v1.9.3, the container fails to start with the following error:

```
panic: constraint failed: NOT NULL constraint failed: ApiKeys.LastUsedString (1299)
```

I have read the breaking changes regarding the API but I cannot understand the correlation and what action I need to take when updating to 1.9.3 to get it working.

Log file (pulled from dozzle)
[gokapi.log](https://github.com/user-attachments/files/18048015/gokapi.log)

config.json (secure elements I have **** out)
```
{
  "Authentication": {
    "Method": 0,
    "SaltAdmin": "***",
    "SaltFiles": "****",
    "Username": "****",
    "Password": "****",
    "HeaderKey": "",
    "OauthProvider": "",
    "OAuthClientId": "",
    "OAuthClientSecret": "",
    "OauthUserScope": "",
    "OauthGroupScope": "",
    "OAuthRecheckInterval": 168,
    "HeaderUsers": null,
    "OAuthGroups": null,
    "OauthUsers": null
  },
  "Port": ":53842",
  "ServerUrl": "https://files.*****.com/",
  "RedirectUrl": "https://www.cloudflarestatus.com/",
  "PublicName": "Files",
  "DataDir": "data",
  "DatabaseUrl": "sqlite://data/gokapi.sqlite",
  "ConfigVersion": 21,
  "LengthId": 15,
  "MaxFileSizeMB": 102400,
  "MaxMemory": 50,
  "ChunkSize": 45,
  "MaxParallelUploads": 4,
  "Encryption": {
    "Level": 0,
    "Cipher": null,
    "Salt": "",
    "Checksum": "",
    "ChecksumSalt": ""
  },
  "UseSsl": false,
  "PicturesAlwaysLocal": false,
  "SaveIp": false,
  "IncludeFilename": false
}
```

Please let me know if you need any further information to assist with diagnosis.

Thanks.

**Comment by Forceu** [comment]
Thanks, I will look into it

**Comment by Forceu** [comment]
Thanks, I will look into it. Did you update from 1.9.2 or from a previous version?

**Comment by beastarman74** [comment]
Updated from 1.9.2

**Comment by Forceu** [comment]
It appears, that for some reason you are still using an old database table, which should have been updated. I pushed a fix, it might take about 30 minutes until the docker image [is built](https://github.com/Forceu/Gokapi/actions/runs/12213004417/job/34072127534).

Once it's built, can you check with the docker tag gokapi:latest-dev if this fixes your problem?

**Comment by beastarman74** [comment]
Sorry for the delay in replying. Yes the latest-dev build fixes the issue.

I have been using Gokapi for sometime. The only thing I can think of is
that up until the last month I was running the latest-dev build, then
switched to the numbered releases to allow control of updates.

Thanks for your help in resolving this so quickly.

**Comment by Forceu** [comment]
Good to hear it is working. To be honest, I am quite surprised, that this condition was possible, I have no idea how it happened. 

Closed in [be589fc](https://github.com/Forceu/Gokapi/commit/be589fc75880bdf4861dccbd9a73dbb4a25f7660)
